skfda/exploratory/stats/_functional_transformers.py

```diff
@@ -75,3 +75,136 @@ def local_averages(
+    intervals_x_inside = np.delete(intervals_x_inside, 0, axis=1)
+
+    return np.sum(intervals_x_inside, axis=1)
+
+
+def occupation_measure(
+    data: Union[FDataGrid, FDataBasis],
+    intervals: np.ndarray,
+    *,
+    n_points: Union[int, None] = None,
+) -> np.ndarray:
+    r"""
+    Calculate the occupation measure of a grid.
+
+    It performs the following map.
+        ..math:
+            :math:`f_1(X) = |t: X(t)\in T_p|,\dots,|t: X(t)\in T_p|`
+
+        where :math:`{T_1,\dots,T_p}` are disjoint intervals in
+        :math:`\mathbb{R}` and | | stands for the Lebesgue measure.
+
+        Args:
+            data: FDataGrid or FDataBasis where we want to calculate
+            the occupation measure.
+            intervals: ndarray of tuples containing the
+            intervals we want to consider. The shape should be
+            (n_sequences,2)
+            n_points: Number of points to evaluate in the domain.
+            By default will be used the points defined on the FDataGrid.
+            On a FDataBasis this value should be specified.
+        Returns:
+            ndarray of shape (n_intervals, n_samples)
+            with the transformed data.
+
+    Example:
+        We will create the FDataGrid that we will use to extract
+        the occupation measure
+        >>> from skfda.representation import FDataGrid
+        >>> import numpy as np
+        >>> t = np.linspace(0, 10, 100)
+        >>> fd_grid = FDataGrid(
+        ...     data_matrix=[
+        ...         t,
+        ...         2 * t,
+        ...         np.sin(t),
+        ...     ],
+        ...     grid_points=t,
+        ... )
+
+        Finally we call to the occupation measure function with the
+        intervals that we want to consider. In our case (0.0, 1.0)
+        and (2.0, 3.0). We need also to specify the number of points
+        we want that the function takes into account to interpolate.
+        We are going to use 501 points.
+        >>> from skfda.exploratory.stats import occupation_measure
+        >>> np.around(
+        ...     occupation_measure(
+        ...         fd_grid,
+        ...         [(0.0, 1.0), (2.0, 3.0)],
+        ...         n_points=501,
+        ...     ),
+        ...     decimals=2,
+        ... )
+        array([[ 1.  ,  0.5 ,  6.27],
+               [ 0.98,  0.48,  0.  ]])
+
+    """
+    if isinstance(data, FDataBasis) and n_points is None:
+        raise ValueError(
+            "Number of points to consider, should be given "
+            + " as an argument for a FDataBasis. Instead None was passed.",
+        )
+
+    for interval_check in intervals:
+        y1, y2 = interval_check
+        if y2 < y1:
+            raise ValueError(
+                "Interval limits (a,b) should satisfy a <= b. "
+                + str(interval_check) + " doesn't",
+            )
+
+    if n_points is None:
+        function_x_coordinates = data.grid_points[0]
```

Comment:
As now, the code only works for R to R functions. Either extend it to arbitrary dimensions or put a check at the beginning to not give bogus results for other functions.

Reply:
You have the function `check_is_univariate` in `_utils` for now, to do this type of checking.